src/model.rs

```diff
@@ -118,4 +121,14 @@ mod test {
         assert!(a.can_upgrade_to(&b));
         assert!(!b.can_upgrade_to(&a));
     }
+
+    /// Validate we're not breaking the serialized format.
+    #[test]
+    fn test_deserialize() -> Result<()> {
+        let data = include_str!("example-state-v0.json");
```

Comment:
I'd suggest that we introduc a `tests/fixtures/` directory and move this one there. It would hopefully encourage a trend of adding more pristine fixtures to cover the interop surface.

Reply:
fixtured.